### DISCUSSION TITLE
NextJS Image lowers uploaded image quality

### DISCUSSION THREAD

**Discussion by IvonaPilcevic** [vercel/vercel#5748] [Help] [answered]
Hi, I am getting my image from api ( this is path for image: https://vvapi.test.tbstaging.dev/storage/600af708e75249.20195232/600af708e75249.20195232.jpeg ), and when I show it ( https://vv.test.tbstaging.dev/rs ) the quality of image gets bad and it kinda gets blurry. I already added quality={100} and it made it a little bit better, but not as clear as it should be. I also tried with loading="eager" and unopitmized={true} and it didn't change anything. My image also has the priority property set to true if it means anything.
I have no idea how to fix this problem, and it's hero image so it's very obvious :) Please help

**Comment by styfle** [comment, marked as answer]
It looks like you're not using the `next/image` component because the image is served as-is.

```html
<img alt="Hero Slider" src="https://vvapi.test.tbstaging.dev/storage/600af708e75249.20195232/600af708e75249.20195232.jpeg" class="css-1x8in2u">
```

Are you sure the original image quality is any better?

Perhaps you can upload the jpeg here can compare.

**Reply by IvonaPilcevic** [reply]
I changed it to img for a few minutes to see if there is any difference. I updated it and now it's using next/image. 
![slider_desktop](https://user-images.githubusercontent.com/26251410/105725977-dd896900-5f29-11eb-9766-b06c7ace1788.jpg) Here is the image I was talking about, it's large and clear and on my server it's all blurry

**Reply by styfle** [reply]
Ok I'm seeing the optimized image now.

https://vv.test.tbstaging.dev/_next/image?url=https%3A%2F%2Fvvapi.test.tbstaging.dev%2Fstorage%2F600af708e75249.20195232%2F600af708e75249.20195232.jpeg&w=1200&q=100


Looks like the width is maxed out at 1200 pixels when it should serve at the device's width.

Did you modify `images` in `next.config.js` or perhaps the `sizes` prop on the component? Try using the defaults instead and see if that helps.

**Reply by IvonaPilcevic** [reply]
I only have domains: [ example.com ] in my _next.config.js,_ and I added property _unsized_ to image, extended it and then gave it width. So my Image component looks like this: 

`export const SliderImage = styled(Image)'
  height: 100%;
  width: 100%;
  max-height: 100%;'

<SliderImage
    priority
    unsized
    quality={100}
    src={
         (isMobile
         ? slide?.image?.mobile?.image_url
         : slide?.image?.desktop?.image_url) || ""
         }
     alt="Hero Slider"
            />
;`

I don't have prop sizes on component or imageSizes in next.config.js ...

**Reply by styfle** [reply]
Looks like you're still on Next 10.0.0

Please update to the latest version (currently 10.0.5).

Then you can remove `unsized` and your CSS in favor of `layout=fill` and `objectFit`.

Try this demo: https://image-component.nextjs.gallery/layout-fill
Source code: https://github.com/vercel/next.js/blob/canary/examples/image-component/pages/layout-fill.js

**Reply by IvonaPilcevic** [reply]
Hi, I updated next to 10.0.5, and now my image looks like this:
 `<Image
              priority
              quality={100}
              width={isMobile ? 750 : 1920}
              height={isMobile ? 1686 : 950}
              layout="responsive"
              objectFit="cover"
              objectPosition="center"
              src={
                (isMobile
                  ? slide?.image?.mobile?.image_url
                  : slide?.image?.desktop?.image_url) || ""
              }
              alt="Hero Slider"
            />`
I didn’t use layout="fill", because we use two images, one for desktop and one for mobile.
I think it is a little bit better, but not good enough as you can see. Is there anything else I can do to improve image quality ? 
Thank you for helping me!

**Reply by styfle** [reply]
The quality looks nearly identical between the original source image and the optimized image.

I think you need to fix your source image.

- [Source Image](https://vvapi.test.tbstaging.dev/storage/600af708e75249.20195232%2F600af708e75249.20195232.jpeg) - 770KB jpeg
- [Optimized Image](https://vv.test.tbstaging.dev/_next/image?url=https%3A%2F%2Fvvapi.test.tbstaging.dev%2Fstorage%2F600af708e75249.20195232%2F600af708e75249.20195232.jpeg&w=3840&q=100) - 248KB webp

Try using a png instead and see if that makes a difference.

**Reply by IvonaPilcevic** [reply]
Sure, I will try to fix it, thaks again for your help and quick answers :)
